### DISCUSSION TITLE
Move the database file to a different location outside of the content folder.

### DISCUSSION THREAD

**Discussion by pandafeliz** [Difegue/LANraragi#487] [Q&A]
Running LANraragi 0.7.9 on Windows 10.

Currently I have all my files in a regular HDD (path x:\xxx\xxx)
The database.rdb file is located in that folder and all my files in different sub folders from that root, but I want to move the database file outside of that folder into a SSD where all the thumbnails are being stored.

I tried using symlink the folders into the new database location (d:\xxx\) but it will just ignore all the previous paths and treat it as a clean installation. 

Thanks for any input on how to accomplish this.

**Comment by Difegue** [comment]
Did you use WSL to create the symlinks? AFAIK regular Windows NTFS junctions should work perfectly fine to move the database to a SSD.  
https://www.2brightsparks.com/resources/articles/ntfs-hard-links-junctions-and-symbolic-links.html

**Reply by pandafeliz** [reply]
This is how my installation looks right now, the client is installed in C:\, which is the SSD alongside the thumbnails folder. My files are located in X:\xxx\manga\, inside there is the rdb file alongside the subfolders where everything is categorized by site.

I tried both ways, first replicating the folder structure I have in my regular hard drive onto the SSD using junctions and putting the db file there, but when I run the client it will throw a message that all my files are in the filesystem but not in the filemap, so if I run the Clean database command, it will wipe the files in the previous location, including all the metadata, and will try to scrap everything again.

Then, doing a symlink of the rdb file only, putting the original in the SSD and then the link in the hard drive, but in that case, when I run the client it will overwrite the link with the original file back in the hard drive and use that, so is back to square one.

Don't know how hard would be to add an option to specify a different location for the database file like you did with the thumbnails folder. Before you added that I actually used a junction to store all the thumbnails in my SSD.

**Reply by Difegue** [reply]
It wouldn't be hard per se to add a third location field for the DB, but I've been trying to avoid overcomplexifying the windows app's settings -- If junctions don't work for the rdb tho it's probably worth adding.   

Maybe for the Windows 11 refresh!
